### PR TITLE
Update vendor'ed code

### PR DESCRIPTION
**Pull request by cfergeau**
This needs an update of the go version that we use.

### CODE REVIEW

**Comment by cfergeau** [comment]
`Switch to golang 1.18` is slightly problematic as `registry.access.redhat.com/ubi8/go-toolset` which is used by `images/Dockerfile` only has golang up to 1.17 :(

File: go.mod [hunk 1]
```diff
@@ -1,13 +1,42 @@
...
+	google.golang.org/protobuf v1.25.0 // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	k8s.io/api v0.21.1 // indirect
+	k8s.io/apimachinery v0.21.1 // indirect
+	k8s.io/klog/v2 v2.8.0 // indirect
+	k8s.io/utils v0.0.0-20201110183641-67b214c5f920 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.1.0 // indirect
+	sigs.k8s.io/yaml v1.2.0 // indirect
+)
```
Comment (praveenkumar) [inline go.mod:53]: @cfergeau Is this generated automatic just after you updated the go version and run `go mod tidy` or part of it done manually?

Comment (cfergeau) [reply go.mod:53]: it was automatic after changing `go 1.15` to `go 1.18` and running `go mod tidy`

File: go.mod [hunk 2]
```diff
@@ -1,13 +1,53 @@
 module github.com/code-ready/routes-controller
 
-go 1.15
+go 1.18
```
Comment (praveenkumar) [inline go.mod:3]: 👍🏼

**Review by praveenkumar — APPROVED** [review]

**Pull request merged**